### PR TITLE
Brian Kernighan's Algorithm

### PR DESCRIPTION
**Pull request by 1vivekanand**
fixed - #44

### CODE REVIEW

File: Code/Java/BrianKernighanAlgorithm.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+		
+		// calling the countSetBits function which will return the number of set bits and storing the result
+		// in a variable called answer
+		int answer = countSetBits(digit);
+		
+		// printing out the output
+		System.out.println("The number of set bits in "+ digit + " is: "+ answer);
+
+	}
+
```
Comment (raksha009) [inline Code/Java/BrianKernighanAlgorithm.java:72]: Remove extra blank line

File: Code/Java/BrianKernighanAlgorithm.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+}
+
+/*
+ 	Test Cases:
+ 		1. input  = 55
```
Comment (raksha009) [inline Code/Java/BrianKernighanAlgorithm.java:77]:
```suggestion
 		1. Input- 55

```

File: Code/Java/BrianKernighanAlgorithm.java [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+ 		   output = 5
```
Comment (raksha009) [inline Code/Java/BrianKernighanAlgorithm.java:78]:
```suggestion
 		   Output- 5

```

File: Code/Java/BrianKernighanAlgorithm.java [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+ 		   
+ 		2. input  = 145
+ 		   output = 3
+ 		   
+ 	Time Complexity  : O(number of set bits in the input)
+ 	Space complexity : O(1)
+  
+ * */
```
Comment (raksha009) [inline Code/Java/BrianKernighanAlgorithm.java:86]:
```suggestion
  */

```

**Review by raksha009 — CHANGES_REQUESTED** [review]

**Comment by 1vivekanand** [comment]
updated the PR

**Review by raksha009 — APPROVED** [review]

**Comment by Simi06** [comment]
Is this issue free to work on??
i would love to contribute

**Comment by 1vivekanand** [comment]
@plazzy99 please review the pr #1244

File: Algorithm/Bit_manipulation/readme.md [hunk 5]
```diff
@@ -52,4 +52,8 @@ There are different bitwise operations used in the bit manipulation. Bit operati
 
 
 ## Questions :
-- Brian Kernighan's Algorithm ----> [C++](/Code/C++/Brian_Kernighan.cpp) 
\ No newline at end of file
+
```
Comment (raksha009) [inline Algorithm/Bit_manipulation/readme.md:55]: Plz bring it back to the original approved state.

**Review by raksha009 — CHANGES_REQUESTED** [review]

File: Algorithm/Bit_manipulation/readme.md (hunk 5 above)
Comment (raksha009) [reply Algorithm/Bit_manipulation/readme.md:55]: Do this change, we'll merge your PR then.

Comment (1vivekanand) [reply Algorithm/Bit_manipulation/readme.md:55]: Done

File: Algorithm/Bit_manipulation/readme.md [hunk 6]
```diff
@@ -52,4 +52,7 @@ There are different bitwise operations used in the bit manipulation. Bit operati
...
+* Brian Kernighan's algorithm --> [Java](/Code/Java/BrianKernighanAlgorithm.java)
+- Brian Kernighan's Algorithm ----> [C++](/Code/C++/Brian_Kernighan.cpp)
```
Comment (raksha009) [inline Algorithm/Bit_manipulation/readme.md:57]: heyy, this has to be written like
```suggestion
- Brian Kernighan's Algorithm ----> [C++](/Code/C++/Brian_Kernighan.cpp) | [Java](/Code/Java/BrianKernighanAlgorithm.java)
```
and then delete line 56

**Review by raksha009 — CHANGES_REQUESTED** [review]

File: Algorithm/Bit_manipulation/readme.md (hunk 6 above)
Comment (1vivekanand) [reply Algorithm/Bit_manipulation/readme.md]: Done

**Review by raksha009 — APPROVED** [review]

**Pull request merged**